hooks/habrok.py

```diff
@@ -112,6 +113,15 @@ def pre_configure_hook(self, *args, **kwargs):
         self.log.info("[pre-configure hook] Updated list of excluded tests: " + ', '.join(self.cfg['excluded_tests']['']))
 
 
+def pre_fetch_hook(self, *args, **kwargs):
+    # Apptainer: set the correct installation directory
+    if self.cfg['easyblock'] == "Apptainer":
+        self.log.info("[pre-install hook] Checking and setting the container installation path")
+        if 'EB_HABROK_CONTAINER_PATH' not in os.environ:
+            raise EasyBuildError('Apptainer builds require $EB_HABROK_CONTAINER_PATH to be set!')
+        self.installdir = os.getenv('EB_HABROK_CONTAINER_PATH') + '/' + self.name + '/' + self.version
```

Comment:
This doesn't really work well, as it will only change the installation directory to the intended location, but the module file will still end up in the original installation path used by EB.

I looked for a way to change EB's installation prefix, and though that's not really supported / easily possible, you can override it in a bit of a hacky way, and I think you can do this is the `parse_hook` as well:

```
    if self['easyblock'] == "Apptainer":
        self.log.info("[pre-install hook] Checking and setting the container installation path")
        if 'EB_HABROK_CONTAINER_PATH' not in os.environ:
            raise EasyBuildError('Apptainer builds require $EB_HABROK_CONTAINER_PATH to be set!')
        config.ConfigurationVariables()._FrozenDict__dict['installpath'] = os.getenv('EB_HABROK_CONTAINER_PATH')
```

This `installpath` is used by EB to set up the real installation directory, both for the software and module file, and it takes the naming scheme into account as well.

Reply:
This would mean that containers are in the end installed in <provided path>/software/<name>/<version>. 

Do we prefer this to either <provided path>/containers/<name>/<version> or just <provided path>/<name>/<version>?

If we prefer any of the two alternatives I can look into what to change to achieve them while also storing the modules.